File: Frontend/src/types/report.ts
export interface AccountBalance {
  id: number
  code: string
  name: string
  balance: string
}

export interface BalanceSheet {
  as_of: string
  assets: AccountBalance[]
  liabilities: AccountBalance[]
  equity: AccountBalance[]
  net_income: string
  total_assets: string
  total_liabilities: string
  total_equity: string
  is_balanced: boolean
  difference: string
}

export interface ProfitLossLine {
  id: number
  code: string
  name: string
  amount: string
}

export interface ProfitLoss {
  date_from: string
  date_to: string
  income: ProfitLossLine[]
  expenses: ProfitLossLine[]
  other_expenses: ProfitLossLine[]
  total_income: string
  total_expenses: string
  total_other_expense: string
  net_profit: string
}

export type DrillDownDocType = 'PURCHASE_ORDER' | 'VENDOR_BILL' | 'SALES_ORDER' | 'CUSTOMER_INVOICE'

export interface BudgetDrillDownItem {
  id: number
  doc_type: DrillDownDocType
  doc_number: string
  doc_date: string
  partner_name: string
  total_amount: string
}

File: Backend/app/schemas/report.py
"""Balance Sheet / P&L response shapes — design doc §4.1, §4.2."""
from datetime import date

from pydantic import BaseModel

from app.schemas.common import Money


class AccountBalanceOut(BaseModel):
    id: int
    code: str
    name: str
    balance: Money


class BalanceSheetOut(BaseModel):
    as_of: date
    assets: list[AccountBalanceOut]
    liabilities: list[AccountBalanceOut]
    equity: list[AccountBalanceOut]
    net_income: Money
    total_assets: Money
    total_liabilities: Money
    total_equity: Money
    is_balanced: bool
    difference: Money


class ProfitLossLineOut(BaseModel):
    id: int
    code: str
    name: str
    amount: Money


class ProfitLossOut(BaseModel):
    date_from: date
    date_to: date
    income: list[ProfitLossLineOut]
    expenses: list[ProfitLossLineOut]
    other_expenses: list[ProfitLossLineOut]
    total_income: Money
    total_expenses: Money
    total_other_expense: Money
    net_profit: Money
